CHANGELOG.md

```diff
@@ -40,11 +40,45 @@ Bottom level categories:
 
 ## Unreleased
 
+## Major changes
+
+### The `diagnostic(…);` directive is now supported in WGSL
+
+Naga now parses `diagnostic(…);` directives according to the WGSL spec. This
+allows users to control certain lints, similar to Rust's `allow`, `warn`, and
+`deny` attributes. For example, this snippet would emit a uniformity error:
+
+```wgsl
+@compute
+fn vs_main() {
+  TODO // Emits a derivative uniformity error during validation.
```

Comment:
@gfx-rs/naga: Do we have an example of code that should fire a uniformity analysis error here?

Reply:
[This one](https://gpuweb.github.io/gpuweb/wgsl/#example-e515f7e4) might be better:

```wgsl
diagnostic(off,derivative_uniformity);
var<private> d: f32;
fn helper() -> vec4<f32> {
  if (d < 0.5) {
    // The derivative_uniformity diagnostic is disabled here
    // by the global diagnostic filter.
    return textureSample(t,s,vec2(0,0));
  } else {
    // The derivative_uniformity diagnostic is set to 'warning' severity.
    @diagnostic(warning,derivative_uniformity) {
      return textureSample(t,s,vec2(0,0));
    }
  }
  return vec4(0.0);
}
```

Reply:
Neither of those are rejected on basis of derivative uniformity in Naga currently, so they're not super workable. 🙁 The second example doesn't even compile in Tint; neither `t` nor `s` are defined.

I'll see if I can make my own example. LMK if you have other ideas, though!

Reply:
Pushed a dissatisfying "here's a snippet that standard WGSL wouldn't accept" snippet that I derived from a CTS test case with 02daf28.

If we can find one that actually gets rejected by Naga later, then great, we can edit the `CHANGELOG` then.